cmd/controller-manager/app/controller-manager.go

```diff
@@ -117,6 +119,25 @@ func Run(opts *options.Options) error {
 		klog.Fatalf("Invalid Feature Gate: %v", err)
 	}
 
+	// validate if a federation namespace is configured
+	if len(opts.Config.FederationNamespace) == 0 {
```

Comment:
This block needs to appear after the block that tries to discover the namespace to ensure that discovery can be attempted if the namespace is not otherwise provided.

Reply:
Fixed.